R/module_teal_with_splash.R

```diff
@@ -95,15 +106,89 @@ srv_teal_with_splash <- function(id, data, modules, filter = teal_slices()) {
           )
         }
       })
-
-      if (!is.reactive(raw_data)) {
-        stop("The delayed loading module has to return a reactive object.")
-      }
       raw_data
     }
 
-    res <- srv_teal(id = "teal", modules = modules, raw_data = raw_data, filter = filter)
-    logger::log_trace("srv_teal_with_splash initialized module with data { toString(get_dataname(data))}.")
+    if (!is.reactive(raw_data)) {
+      stop("The `teal_data_module` must return a reactive expression containing a `teal_data` object.", call. = FALSE)
+    }
+
+    raw_data_checked <- reactive({
+      # custom module can return error
+      data <- tryCatch(raw_data(), error = function(e) e)
+
+      # there is an empty reactive event on init!
+      if (inherits(data, "shiny.silent.error") && identical(data$message, "")) {
+        return(NULL)
+      }
+
+      # to handle qenv.error
+      if (inherits(data, "qenv.error")) {
+        validate(
+          need(
+            FALSE,
+            paste(
+              "Error when executing `teal_data_module`:\n ",
+              data$message,
```

Comment:
I'm not 100% sure whether error message is one string only. For most of the cases it's true but not sure if true for all possible cases. Would be safer to paste-collapse this as well.

Reply:
You're right, `simpleCondition(c("a", "b"))$message` returns a vector. Will fix